packages/data-stores/src/domain-suggestions/index.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { controls } from '@wordpress/data-controls';
+import { registerStore } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import reducer, { State } from './reducer';
+import * as actions from './actions';
+import * as resolvers from './resolvers';
+import * as selectors from './selectors';
+import { DispatchFromMap, SelectFromMap } from '../mapped-types';
+
+export * from './types';
+export { State, STORE_KEY };
+
+export function register() {
+	registerStore< State >( STORE_KEY, { actions, controls, reducer, resolvers, selectors } );
+}
+
+declare module '@wordpress/data' {
+	function dispatch(
+		key: typeof import('./constants').STORE_KEY
+	): DispatchFromMap< typeof actions >;
+	function select( key: typeof import('./constants').STORE_KEY ): SelectFromMap< typeof selectors >;
+}
```

Comment:
Wondering if we should also leave this to the consumer. I think right now, there's a scenario where the user can import this file (which will lead to the 'side effect' of injecting these type defs into `@wordpress/data`), but not actually register the store -- in which case we'll have made wrong assertions about types in `@wordpress/data`, no? (Specifically, existence of a `DOMAIN_SUGGESTIONS` store -- e.g. when autocompleting `select( '` -- when it's not really there.)

Can we move the `declare module` stuff into `register`, or does that have to be top-level?